resolver.go

```diff
@@ -166,22 +172,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
+	for i := len(addr) - 1; i > 0; i-- {
```

Comment:
Since we are shuffling starting from the last element, the 0th element is already in its final place.